clients/da-vinci-client/src/main/java/com/linkedin/davinci/stats/KafkaConsumerServiceStats.java

```diff
@@ -2,46 +2,86 @@
+import io.tehuti.metrics.stats.Min;
 import io.tehuti.metrics.stats.OccurrenceRate;
 import io.tehuti.metrics.stats.Total;
 import java.util.function.LongSupplier;
 
 
+/**
+ * This class provides the stats for Kafka consumer service per region or per store.
+ * Stats inside this class can either:
+ * (1) Total only: The stat indicate total number of all the stores on this host per region.
+ * (2) Total and Per store only: The stat is registered for each store on this host.
+ */
+
 public class KafkaConsumerServiceStats extends AbstractVeniceStats {
-  private final Sensor pollRequestSensor;
+  private final LongAdderRateGauge pollRequestSensor;
   private final Sensor pollRequestLatencySensor;
   private final Sensor pollResultNumSensor;
-  private final Sensor pollNonZeroResultNumSensor;
+  private final LongAdderRateGauge pollNonZeroResultNumSensor;
 
   private final Sensor pollRequestError;
   private final Sensor consumerRecordsProducingToWriterBufferLatencySensor;
   private final Sensor detectedDeletedTopicNumSensor;
   private final Sensor detectedNoRunningIngestionTopicPartitionNumSensor;
   private final Sensor delegateSubscribeLatencySensor;
   private final Sensor updateCurrentAssignmentLatencySensor;
-  private final Sensor consumerSelectionForTopicError;
   private final Sensor maxPartitionsPerConsumer;
   private final Sensor minPartitionsPerConsumer;
   private final Sensor avgPartitionsPerConsumer;
   private final Sensor getOffsetLagIsAbsentSensor;
   private final Sensor getOffsetLagIsPresentSensor;
   private final Sensor getLatestOffsetIsAbsentSensor;
   private final Sensor getLatestOffsetIsPresentSensor;
+  private final Sensor byteSizeSensor;
+  private final Sensor idleTimeSensor;
 
   public KafkaConsumerServiceStats(
       MetricsRepository metricsRepository,
-      String nameWithKafkaClusterAlias,
-      LongSupplier getMaxElapsedTimeSinceLastPollInConsumerPool) {
-    super(metricsRepository, nameWithKafkaClusterAlias);
+      String storeName,
+      LongSupplier getMaxElapsedTimeSinceLastPollInConsumerPool,
+      KafkaConsumerServiceStats totalStats,
+      Time time) {
+    super(metricsRepository, storeName);
+    /**
+     * Below are the sensors that are recording total stats per region and per-store stats.
+     */
+
+    // The bytes of polled pubsub messages for each poll request
+    byteSizeSensor =
+        registerPerStoreAndTotalSensor("bytes_per_poll", totalStats, () -> totalStats.byteSizeSensor, minAndMax());
+    // the number of messages returned by Kafka consumer poll.
+    pollResultNumSensor = registerPerStoreAndTotalSensor(
+        "consumer_poll_result_num",
+        totalStats,
+        () -> totalStats.pollResultNumSensor,
+        new Avg(),
+        new Total(),
+        new Min());
 
-    pollRequestSensor = registerSensor("consumer_poll_request", new OccurrenceRate());
+    /**
+     * Below are the sensors that are recording total stats per each region
```

Comment:
Hmm, the class constructor no longer take in kafkaClusterAlias, how does these below metrics show per-region stats ?

Reply:
It is going to be per store right, as you pas in store name.

Reply:
You are right, it's per store.. but also note that `total` stats object use the same class `KafkaConsumerServiceStats` as per-store stats object so I had to mix these sensors in the same file. In other words, when use per-region stats, aka `total`, devs use `AggKafkaConsumerServiceStats` to record `total` only. When need to use per-store stats, dev use `aggStats.getStoreStats(storeName).dosomething();`

Reply:
OK got it. Thanks for the explanation.